src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -197,39 +197,30 @@ private World createMatchWorld(String worldName, MapInfo info, TerrainModule ter
   }
 
   /**
-   * Creates a new world, wich must be done on the primary thread. If already running on the primary
-   * thread, calls createWorld, otherwise, starts a sync task to run create world, and waits until
-   * it's done.
+   * Creates a new world, which must be done on the primary thread. If already running on the
+   * primary thread, calls createWorld, otherwise, starts a sync task to run create world, and waits
+   * until it's done.
    *
    * @param creator The creator for the world
    * @return A world if it was able to be created, null otherwise
    */
   private World createWorld(final WorldCreator creator) {
     if (server.isPrimaryThread()) return server.createWorld(creator);
-
-    // FIXME: This is pretty dirty, there must be a better way to run & wait on main
-    final Object LOCK = new Object();
+    final CountDownLatch latch = new CountDownLatch(1);
     final AtomicReference<World> world = new AtomicReference<>();
-    final AtomicBoolean isDone = new AtomicBoolean();
     PGM.get()
         .getServer()
         .getScheduler()
         .runTask(
             PGM.get(),
             () -> {
               world.set(server.createWorld(creator));
-              isDone.set(true);
-              synchronized (LOCK) {
-                LOCK.notifyAll();
-              }
+              latch.countDown();
             });
-    while (!isDone.get()) {
-      synchronized (LOCK) {
-        try {
-          LOCK.wait();
-        } catch (InterruptedException ignore) {
-        }
-      }
+    try {
+      latch.await();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
Logger?

Reply:
This is the main thread, if it's getting interrupted it's likely shutting down the server anyways, but will add the logger

Reply:
`logger.log(Level.SEVERE, "...", e)`

We want to always use the logger for when I add Sentry/other logging support.

Reply:
Had already changed it, was failing to push stuff properly